### PR TITLE
perf(inference): skip terminal RMSNorm+lm_head on intermediate long-prefill chunks

### PR DESCRIPTION
**Pull request by ohdearquant**
## What

For a prompt longer than `max_prefill`, `forward_prefill_batched_chunk` is called once per chunk. In the last-position (`!all_positions`) mode, callers keep **only the final chunk's logits** — yet every chunk still ran the terminal tail: MTP pre-final capture, final RMSNorm, the full-vocab last-token `lm_head` GEMV, top-k, and their GPU readback. For all but the last chunk that output is discarded.

This adds an `emit_logits: bool` flag to `forward_prefill_batched_chunk` that gates **only** that terminal tail. Intermediate chunks still:
- commit their per-layer command buffer (advancing the KV cache and GDN recurrent state), and
- advance the session cursor via `set_position`,

then skip the discarded tail and return an empty `Vec`. The `emit_logits == true` path is byte-for-byte unchanged.

Both callers are threaded through:
- `forward_prefill_impl`: `true` for single/last chunk and for `all_positions`; `is_last` in the chunked last-position loop.
- `forward_prefill_from` (cross-turn prefix-reuse path): same, with `is_last = pos + chunk.len() == start_pos + total`.

## Why it's correct

The terminal tail does not feed the residual stream, KV cache, or GDN recurrent state — those are fully advanced by the per-layer loop that precedes it. Skipping it on a chunk whose logits are discarded changes nothing observable.

Guarded by a new gpu-locked, **mutation-sensitive** regression test (`metal_qwen35_chunked_prefill_emit_logits_skip_correctness`): on a 679-token prompt (> `max_prefill`, so 2 chunks) it asserts the chunked path's returned logits have length `vocab_size`, the argmax matches the `all_positions` final-position argmax, and the session cursor fully advances. Verified mutation-sensitive: forcing `is_last = false` (wrongly skipping the last chunk's tail) makes the test fail; restoring it passes (`argmax_ref = argmax_chunked = 271`, tokens = 679).

## Performance

This removes provably-discarded work; it is not a throughput claim. At n=4096 / max_prefill=512 the saving is ~7 full-vocab `lm_head` GEMVs, on the order of a few ms against a multi-second TTFT (below A/B resolution). There is no Criterion group that drives the real multi-chunk `forward_prefill` path, so no meaningful before/after number exists for it; `make bench-compare` (default CPU groups) is unaffected by this change and shows no change. The value here is the removed waste plus the correctness guard.

## Test

```
cargo test -p lattice-inference --features metal-gpu,f16 --release --lib \
  metal_qwen35_chunked_prefill_emit_logits_skip_correctness
# test result: ok. 1 passed  (argmax_ref=271, argmax_chunked=271, tokens=679)
```

### CODE REVIEW

**Comment by ohdearquant** [comment]
`make bench-compare` (origin/main vs HEAD, default `elementwise_cpu_bench` + `simd` groups): no significant change — every group p > 0.05 except one borderline group at p = 0.05 / −1.4%, which is unrelated CPU-group variance from the two-worktree build (a known false-positive of the two-worktree A/B, disprovable by same-binary re-run). These default groups do not cover the Metal chunked-prefill path this PR touches; as noted in the description, there is no Criterion group that drives real multi-chunk `forward_prefill`, so the correctness guard (the new mutation-sensitive gpu-locked regression test) is the operative evidence, not an A/B number.
